### PR TITLE
Facts hvdc constr bounds

### PR DESCRIPTION
**Pull request by awig**
@ttinoco Please review for changes needed to update elastic acpf.

### CODE REVIEW

**Comment by awig** [comment]
I keep getting that `test_constr_CFUNC`  check on the Jacobian on my mac but  not on Linux. I figured that wasn't related but I could be wrong.

**Comment by ttinoco** [comment]
It should be unrelated. I have been seeing this on travis.

File: tests/test_constraints.py [hunk 1]
```diff
@@ -2394,6 +2412,28 @@ def test_constr_BOUND(self):
...
+                self.assertTrue(facts.has_flags('variable','active power'))
+                self.assertTrue(facts.has_flags('variable','reactive power'))
+                self.assertEqual(u[facts.index_v_mag_s],pf.FACTS_INF_VMAG_S)
+                self.assertEqual(l[facts.index_v_mag_s],0.)
+                self.assertEqual(u[facts.index_v_ang_s],pf.FACTS_INF_VANG_S)
+                self.assertEqual(l[facts.index_v_ang_s],-pf.FACTS_INF_VANG_S)
+                self.assertEqual(u[facts.index_P],pf.FACTS_INF_P)
+                self.assertEqual(l[facts.index_P],-pf.FACTS_INF_P)
+                self.assertEqual(u[facts.index_Q],pf.FACTS_INF_Q)
+                self.assertEqual(l[facts.index_Q],-pf.FACTS_INF_Q)
```
[the remark this answers is not in the document]
Comment (awig) [reply tests/test_constraints.py]: I must not have had a file with a facts device.  I'll need to fine one and rerun.

**Comment by ttinoco** [comment]
I am taking care of it now

**Comment by ttinoco** [comment]
Summary of my changes
- removed debug printing statements
- added 'dc power' variables of VSC to constr_BOUND unittests
- fixed num_vars computation with correct value for facts
- added missing constr_BOUND tests for vsc P_dc and i_dc
- fixed facts tests to use proper facts index var fields and cover all vars: v_mag_s, v_ang_s, P_k, P_m, P_dc, Q_k, Q_m, Q_sh, Q_s
- added missing tests for G row into strings for facts and vsc
- updated test_facts (test_psse_sample_case) to be compatible with latest raw-parser change, which sets Q_max_s/Q_min_s to +-INF when i_max_s is zero (zero means no limits according to raw docs)
- extended test_variable_limit to cover facts and vsc and test new code in "get_var_values" (not needed for elastic, just to cover new code)

Merging to dev now

**Comment by awig** [comment]
Thanks very much!  Sorry for all the mistakes

**Pull request merged**